rpcs3/Emu/RSX/VK/VKGSRender.cpp

```diff
@@ -3373,6 +3373,46 @@ void VKGSRender::flip(int buffer, bool emu_flip)
 		vk::change_image_layout(*m_current_command_buffer, target_image, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, present_layout, range);
 	}
 
+	if (m_frame->screenshot_toggle == true)
+	{
+		m_frame->screenshot_toggle = false;
+
+		const size_t sshot_size = buffer_height * buffer_width * 4;
+
+		const auto device = vk::get_current_renderer();
```

Comment:
VkGSRender holds a reference to the current renderer with the member variable m_device.